folly/container/detail/F14Table.h

```diff
@@ -2586,6 +2586,19 @@ class F14Table : public Policy {
 
 #endif // FOLLY_F14_VECTOR_INTRINSICS_AVAILABLE
 
+namespace {
+    const auto erase_if_impl = [](auto& c, auto predicate) {
+      for (auto i = c.begin(), last = c.end(); i != last;) {
+        if (predicate(*i)) {
```

Comment:
The generated code will be more compact if it is written

    auto prev = i++;
    if (predicate(*prev)) {
      c.erase(prev);
    }

Reply:
Nice! I played around with this myself using GCC and Clang trunk on Godbolt and saw the same results. Thanks for the tip 👍 

I've rewritten the loop to use your suggestion.